src/logger.coffee

```diff
@@ -1,156 +1,146 @@
+					isStdErr: true
+				})
+				@_dropCount = 0
+
+	_setup: _.throttle(->
+		@_req = https.request(@_opts)
+
+		# Since we haven't sent the request body yet, and never will,the
+		# only reason for the server to prematurely respond is to
+		# communicate an error. So teardown the connection immediately
+		@_req.on 'response', (res) =>
+			console.log('LogBackend: server responded with status code:', res.statusCode)
+			@_teardown()
+
+		@_req.on('timeout', => @_teardown())
+		@_req.on('close', => @_teardown())
+		@_req.on 'error', (err) =>
+			console.log('LogBackend: unexpected error:', err)
+			@_teardown()
+
+		# Immediately flush the headers. This gives a chance to the server to
+		# respond with potential errors such as 401 authentication error
+		@_req.flushHeaders()
+
+		# We want a very low writable high watermark to prevent having many
+		# chunks stored in the writable queue of @_gzip and have them in
+		# @_stream instead. This is desirable because once @_gzip.flush() is
+		# called it will do all pending writes with that flush flag. This is
+		# now what we want though. If there are 100 items in the queue we want
+		# to write all of them with Z_NO_FLUSH and only afterwards do a
+		# Z_SYNC_FLUSH to maximize compression
+		@_gzip = zlib.createGzip(writableHighWaterMark: 1024)
+		@_gzip.on('error', => @_teardown())
+		@_gzip.pipe(@_req)
+
+		# Only start piping if there has been no error after the header flush.
+		# Doing it immediately would potentialy lose logs if it turned out that
+		# the server is unavailalbe because @_req stream would consume our
+		# passthrough buffer
+		@_timeout = setTimeout(=>
+			@_stream.pipe(@_gzip)
+		, RESPONSE_GRACE_PERIOD)
+	, COOLDOWN_PERIOD)
+
+	_teardown: ->
+		if @_req isnt null
+			clearTimeout(@_timeout)
+			@_req.removeAllListeners()
+			@_req.on('error', ->)
+			# no-op if pipe hasn't happened yet
+			@_stream.unpipe(@_gzip)
+			@_gzip.end()
+			@_req = null
+
+	_snooze: _.debounce(LogBackend::_teardown, KEEPALIVE_TIMEOUT)
+
+	# Flushing every ZLIB_TIMEOUT hits a balance between compression and
+	# latency. When ZLIB_TIMEOUT is 0 the compression ratio is around 5x
+	# whereas when ZLIB_TIMEOUT is infinity the compession ratio is around 10x.
+	_flush: _.throttle(->
+		@_gzip.flush(zlib.Z_SYNC_FLUSH)
+	, ZLIB_TIMEOUT, leading: false)
+
+	_write: (msg) ->
+		@_snooze()
+
+		if @_req is null
+			@_setup()
+
+		if @_writable
+			@_writable = @_stream.write(JSON.stringify(msg) + '\n')
+			@_flush()
+		else
+			@_dropCount += 1
 
-	logDependent: (msg, { uuid }) =>
-		msg = @_truncateToMaxLength(msg)
-		@apiBinder.logDependent(uuid, msg)
+	log: (msg) ->
+		if @offlineMode or !@publishEnabled
+			return
 
-	log: (message) =>
-		return if @offlineMode or !@publishEnabled or @publishQueue.length >= @maxQueueSize
-		if _.isString(message)
-			message = { message: message }
+		if _.isString(msg)
+			msg = { message: msg }
 
-		message = _.defaults({}, message, {
+		msg = _.defaults({}, msg, {
```

Comment:
You can change this to
```coffee
msg = _.assign({
	timestamp: Date.now()
 	message: ''
 }, msg)
```
to get the same default values/cloning semantics without having to check for each property if it already exists